ratelimit/ratelimit.go

```diff
@@ -362,6 +368,8 @@ type limiter interface {
 
 // contextLimiter extends limiter with an AllowContext method that accepts an additional
 // context.Context, e.g. to support OpenTracing.
+//
+// Deprecated: In favour of limiter since Allow is also deprecated now.
```

Comment:
Can we clean this up?
We don't need contextLimiter if it does not differ from limiter interface and it's internal only so we can do whatever we want.